src/python/scripts/dx-clone-asset

```diff
@@ -88,23 +101,29 @@ def _clone_asset_into_region(region, record_name, asset_properties, asset_file_n
 
     # Now try to run the CLONE_ASSET_APP num_retries + 1 times.
     curr_try = 0
+    record_id = None
     while curr_try <= num_retries:
-        job = CLONE_ASSET_APP.run(
-            executable_input={
-                'asset_properties': json.dumps(asset_properties), 'url': url,
-                'record_name': record_name, 'asset_file_name': asset_file_name},
-            project=project_id)
-        print('{0}: {1}'.format(region, job.get_id()), file=sys.stderr)
+        cmd = ['dx', 'run', CLONE_ASSET_APP_NAME, '--project', project_id, '-iurl=' + url, '-irecord_name=' + record_name]
+        cmd += ['-iasset_file_name=' + asset_file_name, '-iasset_properties=' + json.dumps(asset_properties), '--brief']
+        job = subprocess.check_output(cmd).strip()
+        print('{region}: {job_id}'.format(region=region, job_id=job), file=sys.stderr)
         try:
-            job.wait_on_done()
-            q.put(region)
-            return {region: job.describe()['output']['asset_bundle']}
-        except DXJobFailureError:
-            pass
+            cmd = 'dx wait {job_id} '.format(job_id=job)
+            subprocess.check_output(cmd, shell=True)
+        except subprocess.CalledProcessError:
+            traceback.print_exc()
+
+        cmd = 'dx describe {job_id} --json '.format(job_id=job)
+        job_desc = json.loads(subprocess.check_output(cmd, shell=True).strip())
+
+        if job_desc['state'] == 'done':
+            record_id = job_desc['output']['asset_bundle']
+            return {region: job_desc['output']['asset_bundle']}
```

Comment:
It doesn't look like you are placing the region into our queue if we take this codepath.  Perhaps replace the return statement with a simple break?  That way you have set the record_id, it will exit the while loop, the region will be placed in the queue on line 125 and line 126 will return the record_id set here.